### ISSUE TITLE
Output class indicating whether an image has a caption

### ISSUE THREAD

**Issue by JohnONolan** [TryGhost/Ghost#10445]
I think I have a reasonable case for adding a class of `.kg-card-hascaption` conditionally to a parent card wrapper, when it has a caption.

Example:
![image](https://user-images.githubusercontent.com/120485/52163418-46132600-2714-11e9-926c-8a4e91e52b17.png)

In this example I'm trying to achieve different layouts for different image sizes. The first image is `width-full` - where the figcaption overlaid with position absolute. Easy.

The second image is `width-wide` - which I'm trying to style so that it has room for a caption somewhere *around* it.

After a solid hour or two, though, I don't think this is going to be doable without a class on the parent. I've cycled through absolute, flexbox, grid, and table display modes - and all of them get you close, but none of them can account for a layout which automatically adapts to whether or not a caption is present. 

Short version: All the layout modes require setting fixed proportions to manually "make space" for the caption, with the exception of flexbox -- and flexbox has [a bug](https://stackoverflow.com/questions/30788131/css3-flexbox-maintain-image-aspect-ratio) which destroys image aspect ratio when an `img` is a flex-child.

So I think the only way to reasonably achieve this for the parent to specify whether it has a caption, which would open up far, far more options for how to display images with captions.

An additional benefit here would be that we'd actually double the number of available image styles/sizes, because every size can have a unique appearance with/without caption - which also opens up a lot more flexibility in terms of design.

TLDR: I think cards should output an additional class based on whether or not they contain a caption

/cc @peterzimon @kevinansfield

**Comment by JohnONolan** [comment]
Aside: Can we just take a moment to appreciate the colour palette of these labels

**Comment by kevinansfield** [comment]
Although the change is fairly straightforward it requires the `html` field for every post to be re-generated which is usually something we only do for major version bumps due to the potential impact for sites with many posts.

In the meantime it's possible to add the class using a small but of client-side JS:

```javascript
document.querySelectorAll('figure.kg-card').forEach(function (figure) {
  if (figure.querySelector('figcaption')) {
    figure.classList.add('kg-card-hascaption');
  }
});
```

**Comment by JohnONolan** [comment]
@kevinansfield That's fine. How would I trigger manual re-generation? Just update a post?

**Comment by kevinansfield** [comment]
@JohnONolan the change is now in master so new/re-saved posts will start having the class after the next release.

I've added the full migration to our 3.0 tracking list so I'll close this now.
